feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/view/StakingAmountView.kt

```diff
@@ -27,16 +27,26 @@ class StakingAmountView @JvmOverloads constructor(
     val amountInput: EditText
         get() = stakingAmountInput
 
+    private val defaultBackground = context.getCutCornersStateDrawable(
+        idleDrawable = context.getCutCornerDrawable(
+            R.color.blurColor,
+            R.color.white_40
+        )
+    )
+    private val onFocusBackground = context.getCutCornersStateDrawable(
+        idleDrawable = context.getCutCornerDrawable(
+            R.color.blurColor,
+            R.color.white
+        )
+    )
+
     init {
         View.inflate(context, R.layout.view_staking_amount, this)
 
-        with(context) {
-            background = getCutCornersStateDrawable(
-                idleDrawable = getCutCornerDrawable(
-                    R.color.blurColor,
-                    R.color.white
-                )
-            )
+        background = defaultBackground
+
+        amountInput.setOnFocusChangeListener { _, hasFocus ->
+            background = if (hasFocus) onFocusBackground else defaultBackground
```

Comment:
Hi, thanks for your contribution. In my opinion, it will be better to let android do things related to state management (via  StateListDrawable). It will be also more efficient since the background won't be changed each time - the only the drawable state will be refreshed

Reply:
But this is the only solution that I have been able to achieve the parent View's background change when the EditText focus changes. Your solution did not work for me, since it is impossible to change the drawable state by changing the focus of the EditText. Do you have any other ideas?

Reply:
There is `state_focused` for <selector> and `android.R.attr.state_focused` for both ColorStateList and StateListDrawable. Did you try it?

Reply:
Now I tried to apply the addStatesFromChildren flag, it seems to work, the parent state is updated when the child state changes. Thanks!